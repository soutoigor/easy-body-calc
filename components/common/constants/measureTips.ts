import height from '@/assets/images/height.jpg'
import weight from '@/assets/images/weight.jpg'
import hip from '@/assets/images/hip.jpg'
import neck from '@/assets/images/neck.jpg'
import abdomen from '@/assets/images/abdomen.jpg'

export const measureTips = {
  HEIGHT: {
    TITLE: 'Height',
    IMAGE: height,
    DESCRIPTION: `
      With a measuring tape or a ruler, measure your height in centimeters,
      keep your legs together, your body straight and your head centralized.
    `,
  },
  WEIGHT: {
    TITLE: 'Weight',
    IMAGE: weight,
    DESCRIPTION: `
      Check your weight with a scale, preferably in fasting and with underwear.
    `,
  },
  HIP: {
    TITLE: 'Hip',
    IMAGE: hip,
    DESCRIPTION: `
      With a measuring tape, contour the biggest part of your hip and get the centimeters.
    `,
  },
  NECK: {
    TITLE: 'Neck',
    IMAGE: neck,
    DESCRIPTION: `
      With a measuring tape, contour your neck like a necktie, above the trapezius,
      but careful to not take the trapezius in consideration.
    `,
  },
  ABDOMEN: {
    TITLE: 'Abdomen',
    IMAGE: abdomen,
    DESCRIPTION: `
      Exhale all the air, but don't force the abdomen. And with a measuring tape, contour
      your waist right above your belly.
    `,
  },
}
