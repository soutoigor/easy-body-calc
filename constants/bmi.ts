export const BMI_RESULTS = {
  UNDERWEIGHT: 'underweight',
  NORMAL: 'normal',
  OVERWEIGHT: 'overweight',
  OBESE: 'obese',
  EXTREMELY_OBESE: 'extremely obese',
}
