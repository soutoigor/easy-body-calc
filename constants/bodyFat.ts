export const BODY_FAT_RESULTS = {
  COMPETITOR: 'competitor',
  VERY_GOOD: 'very good shape',
  GOOD: 'good shape',
  IN_AVERAGE: 'in average',
  ABOVE_AVERAGE: 'above average',
  OBESE: 'obese',
}
