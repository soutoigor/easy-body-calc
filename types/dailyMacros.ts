export type TGoal = 'lose' | 'keep' | 'gain'

export interface IGoals {
  LOSE: TGoal
  KEEP: TGoal
  GAIN: TGoal
}

export interface IDailyMacros {
  calories: number
  weight: number
  goal: TGoal
}

export interface IDailyMacrosResult {
  goal: TGoal
  fat: number
  protein: number
  carb: number
  calories: number
}

export type TResultForm = IDailyMacrosResult | null
