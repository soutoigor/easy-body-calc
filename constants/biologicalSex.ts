import { IBiologicalSex } from '@/types/biologicalSex'
import { TSelectItems } from '@/types/selectOption'

export const BIOLOGICAL_SEX: IBiologicalSex = {
  MALE: 'M',
  FEMALE: 'F',
}

export const BIOLOGICAL_SEX_ITEMS: TSelectItems = [
  {
    label: 'Male',
    value: BIOLOGICAL_SEX.MALE,
  },
  {
    label: 'Female',
    value: BIOLOGICAL_SEX.FEMALE,
  },
]
