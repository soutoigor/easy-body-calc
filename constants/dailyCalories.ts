import { IActivities } from '@/types/dailyCalories'

export const ACTIVITY: IActivities = {
  SEDENTARY: 1.2,
  SLIGHTLY: 1.375,
  MODERATELY: 1.55,
  ACTIVE: 1.725,
  VERY: 1.9,
}
