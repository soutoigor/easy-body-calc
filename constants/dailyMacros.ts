import { IGoals } from '@/types/dailyMacros'

export const GOALS: IGoals = {
  LOSE: 'lose',
  KEEP: 'keep',
  GAIN: 'gain',
}
