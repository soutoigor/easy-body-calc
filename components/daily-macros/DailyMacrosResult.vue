<template>
  <div>
    <div class="result">
      <div class="result__item">
        <span class="result__value">{{ result.carb }}</span>
        <span class="result__text">Carbs</span>
      </div>
      <div class="result__item">
        <span class="result__value">{{ result.fat }}</span>
        <span class="result__text">Fat</span>
      </div>
      <div class="result__item">
        <span class="result__value">{{ result.protein }}</span>
        <span class="result__text">Protein</span>
      </div>
      <div class="result__item">
        <span class="result__value">{{ result.calories }}</span>
        <span class="result__text">Calories</span>
      </div>
    </div>
    <p class="result__advice">{{ resultAdvice }}</p>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType, computed } from '@nuxtjs/composition-api'
import { IDailyMacrosResult } from '@/types/dailyMacros'
import { GOALS } from '@/constants/dailyMacros'

const { LOSE, KEEP, GAIN } = GOALS

export default defineComponent({
  props: {
    result: {
      type: Object as PropType<IDailyMacrosResult>,
      required: true,
    },
  },
  setup({ result }) {
    const messageByGoal = {
      [LOSE]:
        'lose weight, at the end of a every week, you have not lose weight, reduce the Carb value in 10% until you reach your goal',
      [KEEP]: 'keep your weight',
      [GAIN]:
        'gain weight, at the end of a every week, you have not, increase the Carb value in 10% until you reach your goal',
    }

    const resultAdvice = computed(
      () => `Reach the macros above everyday to ${messageByGoal[result.goal]}.`
    )

    return {
      resultAdvice,
    }
  },
})
</script>

<style scoped>
.result {
  @apply border border-turbo rounded-lg h-60 w-full;
  @apply flex items-center justify-evenly;
}

.result__item {
  @apply flex flex-col items-center;
}

.result__value {
  @apply text-3xl lg:text-4xl italic font-black text-turbo;
}

.result__text {
  @apply italic capitalize font-bold text-xl lg:text-2xl;
}

.result__advice {
  @apply mt-4 text-xl;
}
</style>
