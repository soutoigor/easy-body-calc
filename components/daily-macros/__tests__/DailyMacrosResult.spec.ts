import { mount } from '@vue/test-utils'
import DailyMacrosResult from '@/components/daily-macros/DailyMacrosResult.vue'
import { TGoal, IDailyMacrosResult } from '@/types/dailyMacros'
import { GOALS } from '@/constants/dailyMacros'

const result: IDailyMacrosResult = {
  goal: GOALS.KEEP,
  fat: 65,
  protein: 130,
  carb: 200,
  calories: 2200,
}

const mountComponent = (goal: TGoal | null = null) =>
  mount(DailyMacrosResult, {
    propsData: {
      result: {
        ...result,
        goal: goal || result.goal,
      },
    },
  })

describe('DailyMacrosResult', () => {
  it('should mount the component', () => {
    const wrapper = mountComponent()

    expect(wrapper.vm).toBeDefined()
  })

  it('should display the macros passed as props', () => {
    const wrapper = mountComponent()

    expect(wrapper.text()).toContain(result.goal)
    expect(wrapper.text()).toContain(result.fat)
    expect(wrapper.text()).toContain(result.carb)
    expect(wrapper.text()).toContain(result.protein)
    expect(wrapper.text()).toContain(result.calories)
  })

  it('should match right advice message for goal "KEEP"', () => {
    const wrapper = mountComponent(GOALS.KEEP)

    expect(wrapper.text()).toMatchSnapshot()
  })
  it('should match right advice message for goal "LOSE"', () => {
    const wrapper = mountComponent(GOALS.LOSE)

    expect(wrapper.text()).toMatchSnapshot()
  })
  it('should match right advice message for goal "GAIN"', () => {
    const wrapper = mountComponent(GOALS.GAIN)

    expect(wrapper.text()).toMatchSnapshot()
  })
})
