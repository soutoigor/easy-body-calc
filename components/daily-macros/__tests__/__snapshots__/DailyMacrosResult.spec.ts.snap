// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyMacrosResult should match right advice message for goal "GAIN" 1`] = `"200 Carbs 65 Fat 130 Protein 2200 Calories Reach the macros above everyday to gain weight, at the end of a every week, you have not, increase the Carb value in 10% until you reach your goal."`;

exports[`DailyMacrosResult should match right advice message for goal "KEEP" 1`] = `"200 Carbs 65 Fat 130 Protein 2200 Calories Reach the macros above everyday to keep your weight."`;

exports[`DailyMacrosResult should match right advice message for goal "LOSE" 1`] = `"200 Carbs 65 Fat 130 Protein 2200 Calories Reach the macros above everyday to lose weight, at the end of a every week, you have not lose weight, reduce the Carb value in 10% until you reach your goal."`;
